### PR TITLE
Dialogs service extended

### PR DESCRIPTION
**Pull request by Synthezator**
### Issues Resolved ### 

- fixes #

### API Changes ###

 
 None

### Platforms Affected ### 

- Core (all platforms)
- iOS
- Android

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs [hunk 1]
```diff
@@ -13,6 +13,8 @@
 using Softeq.XToolkit.WhiteLabel.Threading;
 using Softeq.XToolkit.WhiteLabel.Extensions;
 using UIKit;
+using Softeq.XToolkit.WhiteLabel.Mvvm;
```
Comment (TimofeyBurak) [inline Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs]: Seems like usings are not sorted

File: Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs [hunk 2]
```diff
@@ -72,6 +74,39 @@ public class StoryboardDialogsService : IDialogsService
             return dialogResult.Task;
         }
 
+        public Task<TEnum> ShowDialogAsync<TEnum>(string title,
+            string message,
+            Dictionary<TEnum, DialogOption> actions)
+            where TEnum : Enum
```
Comment (TimofeyBurak) [inline Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs]: Should we really restrict usage to only enums? Seems like any type can be used here

File: Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs [hunk 3]
```diff
@@ -72,6 +74,39 @@ public class StoryboardDialogsService : IDialogsService
...
+        {
+            var dialogResult = new TaskCompletionSource<TEnum>();
+
+            Execute.BeginOnUIThread(() =>
```
Comment (TimofeyBurak) [inline Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs]: Implementation of this method is very much looks like implementation of the default ShowDialogAsync, but this method is more versatile. Maybe we should reuse it?
BTW, seems like default implementation of the ShowDialogAsync might be turned into extension in this case

File: Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs (hunk 2 above)
Comment (Synthezator) [reply Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs]: I thought about that too, will remove it

File: Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs (hunk 3 above)
Comment (wcoder) [reply Softeq.XToolkit.WhiteLabel.iOS/Services/StoryboardDialogsService.cs]: I agree with @TimofeyBurak, related to an issue #226

File: Softeq.XToolkit.WhiteLabel/Navigation/IDialogsService.cs [hunk 4]
```diff
@@ -18,10 +18,18 @@ public interface IDialogsService
...
+        /// Probably this must be refactored to use radiobuttons.
+        /// 
+        /// On iOS there can be only one Cancel option (it will be in the bottom) and several others,
+        /// which are not sorted.
+        /// </summary>
+        Task<TResult> ShowDialogAsync<TResult>(string title,
             string message,
-            Dictionary<TEnum, DialogOption> actions)
-            where TEnum : Enum;
+            Dictionary<TResult, DialogOption> actions);
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel/Navigation/IDialogsService.cs]: I propose to go another way:

A separate class without use as a base for `CommandAction`:

```cs
class DialogOption<T>
{
    public T Key { get;set; }
    public string Title { get; set; }
    public CommandActionStyle CommandActionStyle { get; set; }
}
```

```suggestion
            IList<DialogOption<TResult>> actions
```

File: Softeq.XToolkit.WhiteLabel.Droid/Dialogs/DroidFragmentDialogService.cs [hunk 5]
```diff
@@ -35,7 +40,59 @@ public class DroidFragmentDialogService : IDialogsService
...
+
+                foreach (var action in actions)
+                {
+                    EventHandler<DialogClickEventArgs> onOptionChosen = (o, e) =>
+                    {
+                        tcs.TrySetResult(action.Key);
+                        var alertDialog = (AlertDialog) o;
+                        alertDialog.Dismiss();
+                    };
+                    if (action.Value.CommandActionStyle == CommandActionStyle.Destructive)
```
Comment (wcoder) [inline Softeq.XToolkit.WhiteLabel.Droid/Dialogs/DroidFragmentDialogService.cs]: Can be replaced to `switch`

**Comment by wcoder** [comment]
This PR be closed because not actual after the latest, merged PR.

**Pull request closed without merging**